pyoptsparse/pySNOPT/meson.build

```diff
@@ -0,0 +1,43 @@
+fs = import('fs')
+if fs.is_file('source' / 'snoptc.f')
+  message('SNOPT source code found!')
+  HAS_SNOPT = true
+else
+  message('SNOPT not found!')
+  HAS_SNOPT = false
+endif
+
+if HAS_SNOPT
+  c = run_command('source' / 'grab-all-fortran-files.py', check: true)
+  snopt_source_files = c.stdout().strip().split('\n')
```

Comment:
Can we do this for other optimizers as well? Listing out all the files does not seem like the best approach

Reply:
So this type of approach is discouraged in meson, as explained [here](https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard). I personally think the other optimizers don't have quite enough files to need this, especially since we are providing all the source files in pyOptSparse, so there is little advantage. Plus, we are not likely to be adding new files or anything that would impact this.